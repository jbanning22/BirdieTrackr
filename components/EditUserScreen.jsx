import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  TouchableOpacity,
} from 'react-native';
import React, {useState} from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {useNavigation, CommonActions} from '@react-navigation/native';
import axios from 'axios';

const EditUserScreen = ({navigation}) => {
  const [password, setPassword] = useState('');
  const [userName, setUserName] = useState('');
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');

  const editUser = async () => {
    const token = await AsyncStorage.getItem('token');
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    try {
      const editMeRes = await axios.patch(
        'http://localhost:3000/users',
        {
          password: password,
          userName: userName,
          fistName: firstName,
          lastName: lastName,
        },
        {
          headers: headers,
        },
      );
      console.log('edit user returned data is: ', editMeRes.data);
      navigation.navigate('ProfileLanding');
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <ScrollView contentContainerStyle={styles.box1}>
      <Text style={styles.singUpText}>Edit Profile</Text>
      <TextInput
        placeholder="Username"
        style={styles.emailInput}
        value={userName}
        onChangeText={setUserName}
        clearButtonMode={'always'}
      />
      <TextInput
        placeholder="First Name"
        style={styles.emailInput}
        value={firstName}
        onChangeText={setFirstName}
        clearButtonMode={'always'}
      />
      <TextInput
        placeholder="Last Name"
        style={styles.emailInput}
        value={lastName}
        onChangeText={setLastName}
        clearButtonMode={'always'}
      />
      <TextInput
        placeholder="Password"
        style={styles.loginTextInput}
        value={password}
        onChangeText={setPassword}
        clearButtonMode={'always'}
        secureTextEntry={true}
      />
      <TouchableOpacity style={styles.signUpButton} onPress={editUser}>
        <Text style={styles.textButton}>Edit</Text>
      </TouchableOpacity>
      <Button
        title="Back"
        onPress={() => navigation.navigate('ProfileLanding')}
      />
    </ScrollView>
  );
};

export default EditUserScreen;

const styles = StyleSheet.create({
  box1: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  singUpText: {
    fontSize: 28,
    fontWeight: '500',
    fontFamily: 'Helvetica',
    marginBottom: 80,
  },
  signUpButton: {
    width: 200,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#52BEDB',
    marginTop: 35,
    marginBottom: 15,
  },
  loginTextInput: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    // marginTop: 10,
  },
  emailInput: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    marginBottom: 15,
  },
  textButton: {
    fontSize: 18,
    color: 'white',
  },
});
