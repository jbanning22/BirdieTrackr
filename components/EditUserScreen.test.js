import React from 'react';
import {
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react-native';
import EditUserScreen from './EditUserScreen';
import {useMutation} from '@tanstack/react-query';

jest.mock('@tanstack/react-query', () => ({
  useQueryClient: jest.fn(),
  useMutation: jest.fn(),
}));
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: () => null,
}));
jest.mock('@react-navigation/native-stack');

describe('Edit User Screen', () => {
  beforeEach(() => {
    useMutation.mockReturnValue([
      jest.fn().mockResolvedValue({data: {success: true}}),
      {isLoading: false, isError: false, error: null},
    ]);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should match the snapshot', async () => {
    const {toJSON} = render(<EditUserScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('should display avatar icon', async () => {
    render(<EditUserScreen />);
    expect(screen.queryAllByTestId('avatar-icon')).toBeTruthy();
  });

  it('should render the save button', async () => {
    render(<EditUserScreen />);
    await waitFor(() => {
      expect(screen.getByText('Save')).toBeTruthy();
    });
  });

  it('should update inputs correctly', async () => {
    const {getByTestId} = render(<EditUserScreen />);

    const usernameInput = getByTestId('username');
    const cityInput = getByTestId('city');
    const stateInput = getByTestId('state');
    const passwordInput = getByTestId('password');

    fireEvent.changeText(usernameInput, 'JohnDoe');
    fireEvent.changeText(cityInput, 'New York');
    fireEvent.changeText(stateInput, 'NY');
    fireEvent.changeText(passwordInput, 'password123');

    await waitFor(() => {
      expect(usernameInput.props.value).toBe('JohnDoe');
      expect(cityInput.props.value).toBe('New York');
      expect(stateInput.props.value).toBe('NY');
      expect(passwordInput.props.value).toBe('password123');
    });
  });
});
