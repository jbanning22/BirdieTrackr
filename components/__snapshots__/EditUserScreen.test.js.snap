// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit User Screen should match the snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignContent": "center",
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-evenly",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/images/BasketBackground2.png",
          }
        }
        style={
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "height": "100%",
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        style={
          {
            "alignItems": "center",
            "borderRadius": 50,
            "height": 100,
            "justifyContent": "center",
            "width": 100,
          }
        }
      />
      <View
        style={
          {
            "backgroundColor": "#F9FAFB",
            "borderRadius": 15,
            "elevation": 5,
            "margin": 10,
            "marginTop": 2,
            "padding": 10,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            {
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              User Name
            </Text>
            <TextInput
              clearButtonMode="always"
              onChangeText={[Function]}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="username"
              value=""
            />
          </View>
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              First Name
            </Text>
            <TextInput
              clearButtonMode="always"
              onChangeText={[Function]}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="first-name"
              value=""
            />
          </View>
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              Last Name
            </Text>
            <TextInput
              clearButtonMode="always"
              onChangeText={[Function]}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="last-name"
              value=""
            />
          </View>
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              City
            </Text>
            <TextInput
              clearButtonMode="always"
              onChangeText={[Function]}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="city"
              value=""
            />
          </View>
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              State
            </Text>
            <TextInput
              clearButtonMode="always"
              onChangeText={[Function]}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="state"
              value=""
            />
          </View>
          <View
            style={
              {
                "margin": 2,
              }
            }
          >
            <Text
              style={
                {
                  "color": "black",
                  "fontFamily": "Satoshi-Medium",
                  "fontSize": 12,
                  "margin": 5,
                  "marginBottom": 5,
                  "paddingLeft": 15,
                }
              }
            >
              Password
            </Text>
            <TextInput
              autoCorrect={false}
              clearButtonMode="always"
              onChangeText={[Function]}
              secureTextEntry={true}
              style={
                {
                  "alignSelf": "center",
                  "backgroundColor": "#F2F2F2",
                  "borderRadius": 10,
                  "height": 50,
                  "marginBottom": 10,
                  "padding": 10,
                  "width": 280,
                }
              }
              testID="password"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#2D6061",
            "borderRadius": 14,
            "height": 56,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
            "width": 327,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontFamily": "Satoshi-Medium",
              "fontSize": 18,
              "fontWeight": "500",
            }
          }
        >
          Save
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
